packages/builtin/string.pony

```diff
@@ -978,16 +982,13 @@ actor Main
         while i < _size do
           (let c, let len) = utf32(i.isize())
 
-          if chars.contains(c) then
-            // If we find a delimeter, add the current string to the array.
-            occur = occur + 1
-
-            if (n > 0) and (occur >= n) then
-              break
-            end
-
-            result.push(cur = recover String end)
+          let found = match chars
```

Comment:
Simply a branch instruction is not necessarily cheap. We are writing an application where that branch might get called hundreds of thousands of times a second. I haven't tested but under heavy load, jemc's variant sounds like it will perform much better.